src/com/generic/tracker/WarehouseTracker.java

```diff
@@ -0,0 +1,176 @@
+package com.generic.tracker;
+
+import java.util.HashMap;
+import java.util.Map;
+import com.generic.model.Shipment;
+import com.generic.model.Warehouse;
+/**
+ * This class will be responsible for tracking
+ * a collection of warehouses and the shipments
+ * in them. This will allow for consistency
+ * of objects as a shipment can't exist without
+ * a warehouse. It would provide only the
+ * required functionalities such as
+ * adding shipment, printing warehouse details,
+ * enabling and disabled the freight receipt of a warehouse.
+ * 
+ * 
+ * THE MAJOR REASON I MAKE THE METHODS RETURN SOMETHING IS FOR TESTING
+ * THE ONLY METHOD THAT SHOLD BE ALLOWED TO HAVE 
+ * A VOID RETURN TYPE IN THIS CLASS IS printWarehouseDetails() and addWarehouse
+ * 
+ * @author Seyi Ola
+ *
+ */
+public class WarehouseTracker {
+	
+	private Map<Integer, Warehouse> warehouses; // Stores a collection of warehouses
+											    // mapped by their id 
+	
+	public WarehouseTracker()
+	{
+		// Store the warehouse
+		warehouses = new HashMap<>();	
+	}
+	
+	/**
+	 * /**
+	 * Adds a new warehouse to the warehouse collection.
+	 * If the warehouse already exists, we return false.
+	 * @param mWarehouse warehouse to add.
+	 * @return true if add was successful,
+	 * 		   false if add wasn't.
+	 */
+	public boolean addWarehouse(Warehouse mWarehouse)
+	{
+		boolean added = false;
+		if(warehouses.get(mWarehouse.getWarehouseID()) == null)
+		{
+			warehouses.put(mWarehouse.getWarehouseID(), mWarehouse);
+			added = true;
+		}
+		return added;
+	}
+	
+	/**
+	 * Adds a shipment to a warehouse, if the warehouseID
+	 * does not exist, it returns -1, if
+	 * the shipment could not be added because it's 
+	 * freight receipt has ended, it returns 0,
+	 * and if the shipment was successfully added, it 
+	 * returns 1.
+	 * @param warehouseID the warehouseID
+	 * @param shipment the shipment to add to this warehouse
+	 * @return -1 if ID does not exist
+	 * 		  , 0 if freight receipt is disabled,
+	 * 		  , 1 if successfully added.
+	 * 			
+	 */
+	public int addShipment(int warehouseID, Shipment mShipment)
```

Comment:
If you are going to return an int with a possible result of 3 values,
return an ENUM for self-code documentation.

I would prefer just a boolean value for simplicity.

Reply:
Returns boolean